File: leetcode/896. Monotonic Array/869.cpp
// https://helloacm.com/how-to-check-if-an-array-is-monotonic/
// https://leetcode.com/problems/monotonic-array/

class Solution {
public:
    bool isMonotonicIncreasing(vector<int>& A) {
        for (int i = 1; i < A.size(); ++ i) {
            if (A[i] < A[i - 1]) {
                return false;
            }
        }
        return true;
    }
 
    bool isMonotonicDecreasing(vector<int>& A) {
        for (int i = 1; i < A.size(); ++ i) {
            if (A[i] > A[i - 1]) {
                return false;
            }
        }
        return true;
    }
    
    bool isMonotonic(vector<int>& A) {
        return isMonotonicIncreasing(A) || isMonotonicDecreasing(A);
    }
};

File: leetcode/869. Reordered Power of 2/869.py
# https://helloacm.com/teaching-kids-programming-reordered-power-of-two-rearranging-the-digits/
# https://leetcode.com/problems/reordered-power-of-2/
# MEDIUM, MATH, PERMUTATION

class Solution(object):
    def reorderedPowerOf2(self, N):     
        count = Counter(str(N))
        return any(count == collections.Counter(str(1 << b))
                   for b in range(31))
